### PR TITLE
Gift Entry: Add donation matching modal

### PR DESCRIPTION
**Pull request by psadie-sfdo**
# Critical Changes

# Changes
- Adds components for the 'Donation Matching' modal
# Issues Closed

# Community Ideas Delivered

# New Metadata
- Custom Labels
  - commonNo
  - common Yes
  - geBodyMatchingApplyNewPayment
  - geBodyMatchingNewOpportunity
  - geBodyMatchingNoPayments
  - geBodyMatchingPendingDonation
  - geBodyMatchingSelectRecord
  - geBodyMatchingUpdatingDonation
  - geButtonMatchingNewOpportunity
  - geButtonMatchingNewPayment
  - geButtonMatchingReviewDonations
  - geButtonMatchingUpdateDonationSelection
  - geButtonMatchingUpdateOpportunity
  - geButtonMatchingUpdatePayment
  - geHeaderMatchingOpportunity
  - geHeaderMatchingPayment
  - geHeaderMatchingReviewDonations
- Lightning Web Components
  - geDonationMatching
  - geDonationMatchingOpportunityCard
  - geDonationMatchingPaymentCard
  - geReviewDonations
  - utilCard
# Deleted Metadata

### CODE REVIEW

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -5018,6 +5034,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <shortDescription>Message about creating a new Opportunity</shortDescription>
+        <value>You are currently creating a new Opportunity.</value>
+    </labels>
+    <labels>
+        <fullName>geBodyMatchingNoPayments</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Message when an Opportunity has no Payments</shortDescription>
+        <value>This opportunity does not have any payments.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <value>This Opportunity does not have any Payments.</value>
```

File: src/labels/CustomLabels.labels [hunk 2]
```diff
@@ -5018,6 +5034,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <protected>true</protected>
+        <shortDescription>Message when an existing gift found</shortDescription>
+        <value>This donor has pending donations.</value>
+    </labels>
+    <labels>
+        <fullName>geBodyMatchingSelectRecord</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Body message in modal to review donations</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Body for modal to review matching Opportunities and Payments</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 3]
```diff
@@ -5098,6 +5162,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Button to delete a section and remove its fields</shortDescription>
         <value>Delete Section and Fields</value>
     </labels>
+    <labels>
+        <fullName>geButtonMatchingNewOpportunity</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Button to create a new Opportunity in modal to review donations</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Button in Review Donations modal to create a new Opportunity</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 4]
```diff
@@ -5098,6 +5162,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <value>Alternatively, create a new Opportunity.</value>
+    </labels>
+    <labels>
+        <fullName>geButtonMatchingNewPayment</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Button to select Opportunity for Payment edits after donation selection</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Button to add a new Payment to an Opportunity</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 5]
```diff
@@ -5098,6 +5162,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <protected>true</protected>
+        <shortDescription>Button to open Review Donations modal and change Opportunity/Payment selection</shortDescription>
+        <value>Select a different donation.</value>
+    </labels>
+    <labels>
+        <fullName>geButtonMatchingUpdateOpportunity</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Gift Update</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Button to select an existing Opportunity to update</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 6]
```diff
@@ -5098,6 +5162,54 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <value>Update this Opportunity</value>
+    </labels>
+    <labels>
+        <fullName>geButtonMatchingUpdatePayment</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Button to select Opportunity for Payment edits during donation selection</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Button to select an existing Payment to update</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 7]
```diff
@@ -5365,6 +5477,32 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Top of all Gift Entry pages</shortDescription>
         <value>Gift Entry</value>
     </labels>
+    <labels>
+        <fullName>geHeaderMatchingReviewDonations</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Header for modal to review donations</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]:
```suggestion
        <shortDescription>Header for modal to review matching Opportunities and Payments</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 8]
```diff
@@ -5365,6 +5477,32 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <value>Review Donations for {0}</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels]: @psadie-sfdo do you need additional explanation here for what {0} represents?

**Comment by melissabarber** [comment]
@psadie-sfdo @ptewson-sfdo @vmartinov-sforg I reviewed the custom labels. There are a few suggestions for updates to shortDescriptions.

**Comment by psadie-sfdo** [comment]
@melissabarber @vmartinov-sforg - updated the PR based on your feedback. Thank you both!

File: src/labels/CustomLabels.labels (hunk 8 above)
Comment (psadie-sfdo) [reply src/labels/CustomLabels.labels]: Yes definitely! Added that in.

**Comment by psadie-sfdo** [comment]
**lurch: attach W-038411

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HqHkQAK" target="_blank">W-038411</a>

**Pull request merged**